src/main/java/networkmanagementsystem/NetworkManagementClient.java

```diff
@@ -5,670 +5,770 @@
-	private static FrequencyBand chooseFreqBand(RatType ratType, Scanner input) {
-		FrequencyBand freqBand = null;
-		int i = 0;
-		System.out.println("Please choose Carrier Frequency Band:\n"
-				         + "Enter 'Back' to go back to the menu\n");
-		if (!ratType.equals(RatType.LTE)) {
-			for (WcdmaFrequencyBand currWcdmaFreq : WcdmaFrequencyBand.values()) {
-				System.out.println("	" + ++i + ".	" + currWcdmaFreq.toString());
-			}
-		} else {
-			for (LteFrequencyBand currLteFreq : LteFrequencyBand.values()) {
-				System.out.println("	" + ++i + ".	" + currLteFreq.toString());
-			}
-		}
-		do {
-			String subOption = input.next();
-			if (subOption.equalsIgnoreCase("BACK")) {
-				break;
-			}
-			try {
-				if (ratType.equals(RatType.LTE)) {
-					freqBand = LteFrequencyBand.values()[Integer.parseInt(subOption) - 1];
-				} else {
-					freqBand = WcdmaFrequencyBand.values()[Integer.parseInt(subOption) - 1];
-				}
-			} catch (ArrayIndexOutOfBoundsException | NumberFormatException ex) {
-				System.out.println("Invalid input, please try again.");
-			}
-		} while (freqBand == null);
-
-		return freqBand;
-	}
+            try {
+                if (i < 4) {
+                    alarm = AlarmStatusLevel.values()[Integer.parseInt(subOption) - 1];
+                } else {
+                    throw new NumberFormatException();
+                }
+            } catch (ArrayIndexOutOfBoundsException | NumberFormatException ex) {
+                System.out.println("Invalid input, please try again.");
+            }
+        } while (alarm == null);
+
+        return alarm;
+    }
+
+    /**
+     * Prepare Observer relations between classes.
+     * Currently, these include the Commission/Decommission classes, the
+     * NetworkManagementSystem, the Mediator, and the NetworkManagementClient.
+     */
+    private static void setupObserverRelations() {
+        // Setup Observer relations
+        MediatorIf mediator = Mediator.getInstance();
+        networkManagementSys = NetworkManagementSystem.getInstance();
+
+        try {
+            Naming.rebind(NETWORK_MGMT_ID, networkManagementSys);
+
+            networkManagementSys.addPropertyChangeListener((PropertyChangeListener) mediator);
+            mediator.addPropertyChangeListener((PropertyChangeListener) networkManagementSys);
+
+            for (Map.Entry<RatType, CommissionRadioUnit> entry : networkManagementSys.getCommissioners().entrySet()) {
+                entry.getValue().addPropertyChangeListener((PropertyChangeListener) mediator);
+            }
+
+            for (Map.Entry<RatType, DecommissionRadioUnit> entry : networkManagementSys.getDecommissioners().entrySet()) {
+                entry.getValue().addPropertyChangeListener((PropertyChangeListener) mediator);
+            }
+        } catch (RemoteException | MalformedURLException ex) {
+            System.out.println("Failed to connect to remote server");
+            ex.printStackTrace();
+            System.exit(0);
```

Comment:
Don't address this now - but I was thinking last week we may not want to exit here if we can't connect to the server. From the perspective of this application, the remote server isn't necessary, and it would be easier to work on issues specific to this part if we didn't need to have rmiregistry up every time. Just leaving this note so that I don't forget to address the mistake I made

Reply:
This is a good point, it would be nice if the entire application didn't fail based on this aha.